File: python3/LeetCode/numRescueBoats.py
from typing import List


class Solution:
    def numRescueBoats(self, people:List[int], limit:int):
        people.sort()
        left=0
        right = len(people)-1
        res=0
        while left<=right:
            if people[left]+people[right]<=limit:
                left+=1
            res+=1
            right-=1
        return res

File: go/LeetCode/numRescueBoats.go
package main

import "sort"

func main() {
	
}

func numRescueBoats(people []int, limit int) int {
	sort.Ints(people)
	left:=0
	right:=len(people)-1
	res:=0
	for left<=right {
		if people[left]+people[right] <= limit {
			left++
		}
		res++
		right--
	}
	return res
}